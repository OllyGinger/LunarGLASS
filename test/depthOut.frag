#version 130

varying vec4 Color;
varying float Depth;

out vec4 foo;

void main()
{
    gl_FragDepth = Depth;
    gl_FragColor = Color;
    foo = Color;
}
