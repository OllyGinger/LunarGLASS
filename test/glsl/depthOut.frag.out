
Top IR:
; ModuleID = 'Top'

@gl_FragDepth = internal global float 0.000000e+00
@gl_FragColor = internal global <4 x float> zeroinitializer
@foo = internal global <4 x float> zeroinitializer

define fastcc void @main() {
entry:
  %Depth = call float @llvm.gla.fReadInterpolant.f32(i32 0, i32 -1, i32 1)
  store float %Depth, float* @gl_FragDepth
  %Color = call <4 x float> @llvm.gla.fReadInterpolant.v4f32(i32 1, i32 -1, i32 1)
  store <4 x float> %Color, <4 x float>* @gl_FragColor
  %Color1 = call <4 x float> @llvm.gla.fReadInterpolant.v4f32(i32 1, i32 -1, i32 1)
  store <4 x float> %Color1, <4 x float>* @foo
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %entry
  %0 = load float* @gl_FragDepth
  call void @llvm.gla.fWriteData.f32(i32 0, i32 -1, float %0)
  %1 = load <4 x float>* @gl_FragColor
  call void @llvm.gla.fWriteData.v4f32(i32 1, i32 -1, <4 x float> %1)
  %2 = load <4 x float>* @foo
  call void @llvm.gla.fWriteData.v4f32(i32 2, i32 -1, <4 x float> %2)
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

declare float @llvm.gla.fReadInterpolant.f32(i32, i32, i32) nounwind readnone

declare <4 x float> @llvm.gla.fReadInterpolant.v4f32(i32, i32, i32) nounwind readnone

declare void @llvm.gla.fWriteData.f32(i32, i32, float) nounwind

declare void @llvm.gla.fWriteData.v4f32(i32, i32, <4 x float>) nounwind


Bottom IR:
; ModuleID = 'Top'

define fastcc void @main() {
entry:
  %Depth = call float @llvm.gla.fReadInterpolant.f32(i32 0, i32 -1, i32 1)
  %Color = call <4 x float> @llvm.gla.fReadInterpolant.v4f32(i32 1, i32 -1, i32 1)
  call void @llvm.gla.fWriteData.f32(i32 0, i32 -1, float %Depth)
  call void @llvm.gla.fWriteData.v4f32(i32 1, i32 -1, <4 x float> %Color)
  call void @llvm.gla.fWriteData.v4f32(i32 2, i32 -1, <4 x float> %Color)
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %entry
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

declare float @llvm.gla.fReadInterpolant.f32(i32, i32, i32) nounwind readnone

declare <4 x float> @llvm.gla.fReadInterpolant.v4f32(i32, i32, i32) nounwind readnone

declare void @llvm.gla.fWriteData.f32(i32, i32, float) nounwind

declare void @llvm.gla.fWriteData.v4f32(i32, i32, <4 x float>) nounwind
compiling depthOut.frag...
varying vec4 Color ; 
varying float Depth ; 
out vec4 foo ; 
void  main (){
gl_FragDepth = Depth ; gl_FragColor = Color ; foo = gl_FragColor ; }



(
(declare (out ) vec4 foo)
(declare (in ) float Depth)
(declare (in ) vec4 Color)
(declare (out ) vec4 gl_FragColor)
(declare (out ) float gl_FragDepth)
(function main
  (signature void
    (parameters
    )
    (
      (assign (constant bool (1)) (x) (var_ref gl_FragDepth)  (var_ref Depth) ) 
      (assign (constant bool (1)) (xyzw) (var_ref gl_FragColor)  (var_ref Color) ) 
      (assign (constant bool (1)) (xyzw) (var_ref foo)  (var_ref Color) ) 
    ))

)


)
// LunarGOO output
#version 130
out vec4 foo;
in float Depth;
in vec4 Color;

void main()
{
    gl_FragDepth = Depth;
    gl_FragColor = Color;
    foo = Color;
    
}
